src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImpl.java

```diff
@@ -5,55 +5,159 @@
+        this(connector,
+             wiresManager,
+             defaultClickHandlers.onClick(),
+             defaultClickHandlers.onDoubleClick());
+    }
+
+    public WiresConnectorHandlerImpl(final WiresConnector connector,
+                                     final WiresManager wiresManager,
+                                     final Consumer<NodeMouseClickEvent> clickEventConsumer,
+                                     final Consumer<NodeMouseDoubleClickEvent> doubleClickEventConsumer) {
         this.m_connector = connector;
-        m_wiresManager = wiresManager;
+        this.m_wiresManager = wiresManager;
+        this.clickEventConsumer = clickEventConsumer;
+        this.doubleClickEventConsumer = doubleClickEventConsumer;
     }
 
     @Override
-    public void onNodeDragStart(NodeDragStartEvent event) {
-        this.m_control.onMoveStart(event.getDragContext().getDragStartX(),
+    public void onNodeDragStart(final NodeDragStartEvent event) {
+        this.getControl().onMoveStart(event.getDragContext().getDragStartX(),
                                    event.getDragContext().getDragStartY());
     }
 
     @Override
-    public void onNodeDragMove(NodeDragMoveEvent event) {
-        this.m_control.onMove(event.getDragContext().getDragStartX(),
+    public void onNodeDragMove(final NodeDragMoveEvent event) {
+        this.getControl().onMove(event.getDragContext().getDragStartX(),
                               event.getDragContext().getDragStartY());
     }
 
     @Override
-    public void onNodeDragEnd(NodeDragEndEvent event) {
-        this.m_control.onMoveComplete();
+    public void onNodeDragEnd(final NodeDragEndEvent event) {
+        this.getControl().onMoveComplete();
     }
 
     @Override
-    public void onNodeMouseClick(NodeMouseClickEvent event) {
-        if (m_wiresManager.getSelectionManager() != null) {
-            m_wiresManager.getSelectionManager().selected(m_connector, event.isShiftKeyDown());
-        }
+    public void onNodeMouseClick(final NodeMouseClickEvent event) {
+       clickEventConsumer.accept(event);
     }
 
     @Override
-    public void onNodeMouseDoubleClick(NodeMouseDoubleClickEvent event) {
-        m_control.addControlPoint(event.getX(), event.getY());
+    public void onNodeMouseDoubleClick(final NodeMouseDoubleClickEvent event) {
+        doubleClickEventConsumer.accept(event);
     }
 
     public WiresConnectorControl getControl() {
-        return m_control;
+        return m_connector.getControl();
+    }
+
+    WiresConnector getConnector() {
+        return m_connector;
+    }
+
+    WiresManager getWiresManager() {
+        return m_wiresManager;
+    }
+
+    public static class DefaultClickHandlers {
+
+        private final Timer clickTimer;
+        private final WiresConnector connector;
+        private final boolean switchVisiblityOnClick;
+        private boolean isShiftKeyDown;
+
+        public DefaultClickHandlers(final WiresManager wiresManager,
```

Comment:
The `DefaultClickHandlers` could have another name that represents the action that will be taken on the ControlPoints, in this case, the DoubleClick -> add and Click -> switchVisiblity. This will be better in case we change the trigger to these actions, instead of double-click to add we could change to another shortcut.

Reply:
Yeah hehe, another good catch! :+1: 
Tbh it was not clear for me neither. 
Just refactored to make it easier to use, the idea now is:
* You can always return a custom handler implementation by overriding the WiresHandlerFactory
* Or you can use this built-in WiresConnectorHandlerImpl by passing the consumer arguments for each event
* Also refactored from lienzo events to some Event class
* Added a built-in "event consumer"'s factory that implementations can use or skip